test/scales/linearScaleTests.ts

```diff
@@ -1,134 +1,267 @@
 
-    it("domainMax()", () => {
-      let scale = new Plottable.Scales.Linear();
-      scale.padProportion(0);
-      let requestedDomain = [-5, 5];
-      scale.addIncludedValuesProvider((scale: Plottable.Scales.Linear) => requestedDomain);
-
-      let maxAboveTop = 10;
-      scale.domainMax(maxAboveTop);
-      assert.deepEqual(scale.domain(), [requestedDomain[0], maxAboveTop], "upper end of domain was set by domainMax()");
-      assert.strictEqual(scale.domainMax(), maxAboveTop, "returns the set maximum value");
-
-      let maxInMiddle = 0;
-      scale.domainMax(maxInMiddle);
-      assert.deepEqual(scale.domain(), [requestedDomain[0], maxInMiddle], "upper end was set even if requested value cuts off some data");
-
-      scale.autoDomain();
-      assert.deepEqual(scale.domain(), requestedDomain, "calling autoDomain() overrides domainMax()");
-      assert.strictEqual(scale.domainMax(), scale.domain()[1], "returns autoDomain()-ed max value after autoDomain()-ing");
-
-      let maxEqualBottom = scale.domain()[0];
-      scale.domainMax(maxEqualBottom);
-      assert.deepEqual(scale.domain(), [maxEqualBottom - 1, maxEqualBottom],
-        "domain is set to [max - 1, max] if the requested value is <= to autoDomain()-ed min value");
-
-      scale.domainMax(maxInMiddle);
-      let requestedDomain2 = [-10, 10];
-      scale.addIncludedValuesProvider((scale: Plottable.Scales.Linear) => requestedDomain2);
-      assert.deepEqual(scale.domain(), [requestedDomain2[0], maxInMiddle], "adding another ExtentsProvider doesn't change domainMax()");
-    });
+        assert.strictEqual(scale.domain([5, 6]), scale, "setting the domain returns the scale");
+        assert.strictEqual(scale.range([6, 7]), scale, "setting the range returns the scale");
+
+        assert.deepEqual(scale.domain(), [5, 6], "the domain has been set");
+        assert.deepEqual(scale.range(), [6, 7], "the range has been set");
+
+        scale.domain([6, 5]);
+        scale.range([7, 6]);
+        assert.deepEqual(scale.domain(), [6, 5], "the domain can be unordered");
+        assert.deepEqual(scale.range(), [7, 6], "the range can be unordered");
+      });
+
+      it("maps domain to range in a linear fashion", () => {
+        scale.domain([10, 20]);
+        scale.range([400, 500]);
+        assert.strictEqual(scale.scale(10), 400, "first value in domain maps to first value in range");
+        assert.strictEqual(scale.scale(20), 500, "last value in domain maps to last value in range");
+        assert.strictEqual(scale.scale(15), 450, "middle value in domain maps to middle value in range");
+
+        assert.strictEqual(scale.invert(400), 10, "first value in range maps to first value in domain");
+        assert.strictEqual(scale.invert(500), 20, "last value in range maps to last value in domain");
+        assert.strictEqual(scale.invert(450), 15, "middle value in range maps to middle value in domain");
+
+        scale.domain([20, 10]);
```

Comment:
Do you think it's worth pulling the reversed-domain tests into a different `it()`?